sirepo/package_data/static/js/radia.js

```diff
@@ -1030,6 +1069,19 @@ SIREPO.app.controller('RadiaSourceController', function (appState, geometry, pan
                         self.selectedObject = null;
                     }
                 }
+                if (o.type === 'racetrack') {
```

Comment:
I don't know radia so take this with a grain of salt. But, I'm struggling to understand what is going on here. I think this whole on model changed function is quite large and could benefit from more functions. At the very least it would be helpful to me to bundle all of the work for o.type === racetrack into a named function. I think since sides and radii aren't single letter variable names it is a tip to me that too much is going on.